io/io/src/TMemFile.cxx

```diff
@@ -134,6 +135,33 @@ TMemFile::TMemFile(const char *path, ExternalDataPtr_t data) :
    Init(!NeedsExistingFile(optmode));
 }
 
+////////////////////////////////////////////////////////////////////////////////////
+/// Constructor to create a read-only TMemFile using an std::unique_ptr<TBufferFile>
+
+TMemFile::TMemFile(const char *name, std::unique_ptr<TBufferFile> buffer) :
+   TFile(name, "WEB", "read-only TMemFile", 0 /* compress */),
+   fBlockList(reinterpret_cast<UChar_t*>(buffer->Buffer()), buffer->BufferSize()),
+   fSize(buffer->BufferSize()), fSysOffset(0), fBlockSeek(&(fBlockList)), fBlockOffset(0)
+{
+   fD = 0;
+   fOption = "READ";
+   fWritable = false;
+
+   // Note: We need to release the buffer here to avoid double delete.
+   // The memory of a TBufferFile is allocated with new, so we can let
```

Comment:
Did you mean 'new []' here ?

Reply:
Yes, I didn't bother to add the bracket, but it's a valid point. I will fix it.

Reply:
Done.